### PR TITLE
Add "Add Last Modified Time" recipe

### PR DESCRIPTION
**Pull request by carsakiller**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- What does this PR change? Give us a brief description.
  - Adds a “Add Last Modified Time” recipe to the documentation, as discussed in #3930. I followed the “reading time” recipe as a guide, as they are very similar.

[Preview article](https://deploy-preview-3991--astro-docs-2.netlify.app/en/recipes/modified-time/)

### CODE REVIEW

**Comment by sarah11918** [comment]
Hey @carsakiller! So happy you added this! Just a quick note because we are a little backed up this week, but it's on the stack! Wanted to make sure you knew it was appreciated and we're trying to get to it... 😅

File: src/content/docs/en/recipes/modified-time.mdx
Comment (ElianCodes) [inline src/content/docs/en/recipes/modified-time.mdx:34]: Just a whitespace nit

```suggestion
   <PackageManagerTabs>
     <Fragment slot="npm">
       ```shell
       npm install dayjs
       ```
     </Fragment>
     <Fragment slot="pnpm">
       ```shell
       pnpm install dayjs
       ```
     </Fragment>
     <Fragment slot="yarn">
       ```shell
       yarn add dayjs
       ```
     </Fragment>
   </PackageManagerTabs>
```

File: src/content/docs/en/recipes/modified-time.mdx
Comment (ElianCodes) [inline src/content/docs/en/recipes/modified-time.mdx:42]:
```suggestion
   :::note
   It's highly recommended to use Git to get the modified time of the files. Git operations (like cloning) will result in the file system modified time being updated.
   :::
```

**Review by ElianCodes — COMMENTED** [review]
Hey @carsakiller thanks for your PR!

In the wait for a final deep review by @sarah11918, I added a couple of small nits. Everything else looks good to me!

Thanks for the recipe ❤️

**Comment by sarah11918** [comment]
Thanks for this, @carsakiller ! 

So, question... I have neither git nor my files locally in a filesystem on my machine (I code in the cloud in Gitpod), so... can I use this recipe?  :smile: 

I was wondering where those things you were importing came from, and whether your Astro project "just knew" about those... and now I think I know why I was confused. Will this work if your git repository/file system is not local on your machine?

File: src/content/docs/en/recipes/modified-time.mdx
Comment (sarah11918) [inline src/content/docs/en/recipes/modified-time.mdx:73]:
``````suggestion
2. Create a Remark Plugin

   This plugin uses `execSync` to run a git command that returns the timestamp of the latest commit in ISO 8601 format. The timestamp is then added to the frontmatter of the file.

     ```js
     import { execSync } from "child_process";

     export function modifiedTime() {
       return function (tree, file) {
         const filepath = file.history[0];
         const result = execSync(`git log -1 --pretty="format:%cI" ${filepath}`);
         file.data.astro.frontmatter.lastModified = result.toString();
       };
     }
     ```
		<details>
		<summary>Using the file system instead of git</summary>
			Although using git is the recommended way to get the last modified timestamp from a file, it is possible to use the file system directly. 
     This plugin uses `statSync` to get the `mtime` (modified time) of the file in ISO 8601 format. The timestamp is then added to the frontmatter of the file.
     ```js
     import { statSync } from "fs";

     export function modifiedTime() {
       return function (tree, file) {
         const filepath = file.history[0];
         const result = statSync(filepath);
         file.data.astro.frontmatter.lastModified = result.mtime.toISOString();
       };
     }
     ```
			</details>
``````

So, this formatting may end up being off, but when one version is *strongly recommended*, it's often better to just show that one. (And, we don't like Notes when we don't really need them.)

So, what about something like *this*? (I'll want to see it in the preview to be sure, but I think this might be a good way of making it clear which one is recommended, and not putting extra info for the people who aren't going to follow your recommendation.)

**Review by sarah11918 — COMMENTED** [review]
I think this looks like a super helpful recipe, @carsakiller !

I got a little creative with just one suggestion idea. (Which may admittedly need some fine-tuning.) What do we all think?

File: src/content/docs/en/recipes/modified-time.mdx
Comment (carsakiller) [reply src/content/docs/en/recipes/modified-time.mdx:73]: Looks good! I'll take a look at some formatting tweaks.

**Review by ElianCodes — COMMENTED** [review]
besides for a very small whitespace nit, LGTM!

**Review by sarah11918 — APPROVED** [review]
I do kind of like the details element, eh?

Great job, @carsakiller ! Appreciate you jumping on this when we discussed the idea!

@ElianCodes happy for you to merge this one when you're happy!

Go Team!  :partying_face:

**Comment by carsakiller** [comment]
Thanks for the help, guys! 🥂🎉

**Comment by ElianCodes** [comment]
Yeessssss! Look good to me!

Let's get this merged! 🚀🛳️

**Pull request merged**
